chain/client/src/sync.rs

```diff
@@ -182,12 +182,16 @@ impl HeaderSync {
                                 if now > *stalling_ts + self.stall_ban_timeout
                                     && *highest_height == peer.chain_info.height
                                 {
-                                    info!(target: "sync", "Sync: ban a fraudulent peer: {}, claimed height: {}, score: {}",
+                                    info!(target: "sync", "Sync: ban a fraudulent peer: {}, claimed height: {}, score: {}", 
                                         peer.peer_info, peer.chain_info.height, peer.chain_info.score);
                                     self.network_adapter.do_send(NetworkRequests::BanPeer {
                                         peer_id: peer.peer_info.id.clone(),
                                         ban_reason: ReasonForBan::HeightFraud,
                                     });
+                                    // This peer is fraudulent, let's skip this beat and wait for
+                                    // the next one when this peer is not in the list anymore.
+                                    self.syncing_peer = None;
+                                    return false;
```

Comment:
It is not really clear what you are doing here.
The boolean here means whether the header sync is due. Returning `false` in the conditions in which it might be due is either a bug or a hack, no?

My understanding of what you attempt to achieve is return false now, so that header sync is not triggered. I presume that you expect that it will still trigger during the next sync beat, at which point you expect that the peer is removed.

I think it is not an acceptable solution for two reasons:
1. There's no guarantee that the peer is removed by the next sync, so you are not solving the problem to begin with? You are just noticeably decreasing the chance of it.
2. It violates the contract of the method. So if we are sticking with this solution, it needs to be documented all the way (here, in the method comment, at the call site, and in the spec).

I would suggest to instead check if our *current* peer is banned, at the beginning of `HeaderSync::run` or `HeaderSync::header_sync_due`, and set it to `None` if it is

Reply:
Actually, turns out I misunderstood what `header_sync_due` return value means. The change is not as hacky as I thought. Comments would still be good :) But seems like we don't really violate the contract of the method with this change, so no actionable item here.

Reply:
I do agree that it actually doesn't remove banned peer - but there is no actual way to know that some peer didn't get banned by any other part of the system. That's why I'm also disabled showing actual error as it's not an error.

Here we for sure know we will try to send it to the peer who is fraudulent (because he has the highest height) - hence skipping this beat.